nuxeo-features/nuxeo-platform-comment/nuxeo-platform-comment/src/main/java/org/nuxeo/ecm/platform/comment/impl/AbstractCommentManager.java

```diff
@@ -193,4 +205,80 @@ protected void notifyEvent(CoreSession session, String eventType, DocumentModel
         notifyEvent(session, eventType, commentParent, commentDocumentModel);
     }
 
+    /**
+     * Checks if a document has comments.
+     *
+     * @param session the core session
+     * @param document the document model who's comments are being counted
+     * @return true if comments were found
+     * @since 11.1
+     */
+    protected boolean hasComments(CoreSession session, DocumentModel document) {
+        // CommentManagerImpl doesn't support it.
+        throw new UnsupportedOperationException();
+    }
+
+    /**
+     * Checks if a document has comments from a particular user.
+     *
+     * @param session the core session
+     * @param document the document model who's comments are being counted
+     * @return true if comments were found
+     * @since 11.1
+     */
+    protected boolean hasCommentsByUser(CoreSession session, DocumentModel document, String author) {
+        // CommentManagerImpl doesn't support it.
+        throw new UnsupportedOperationException();
+    }
+
+    /**
+     * Subscribes top level document's author on first comment and commentor on his first comment if autosubscribtion is
+     * turned on.
+     *
+     * @param session the core session
+     * @param commentDocModel the comment
+     * @param document the document being commented
+     * @since 11.1
+     */
+    protected void handleNotificationAutoSubscriptions(CoreSession session, DocumentModel commentDocModel,
+                                                       DocumentModel document) {
+        if (Framework.getService(ConfigurationService.class).isBooleanFalse(AUTOSUBSCRIBE_CONFIG_KEY)) {
+            return;
```

Comment:
i think a log message on `TRACE` level will help any one trying to understand why the notification mail are received